koku/api/tags/aws/openshift/queries.py

```diff
@@ -28,13 +27,21 @@ class OCPAWSTagQueryHandler(AWSTagQueryHandler, OCPTagQueryHandler):
             "db_table": OCPAWSTagsSummary,
             "db_column_period": "cost_entry_bill__billing_period",
             "annotations": {
-                "enabled": Exists(AWSEnabledTagKeys.objects.filter(key=OuterRef("key")).filter(enabled=True))
+                "enabled": Exists(
+                    EnabledTagKeys.objects.filter(provider_type=Provider.PROVIDER_AWS)
+                    .filter(key=OuterRef("key"))
+                    .filter(enabled=True)
+                )
```

Comment:
Just for future reference, _all_ `.filter().filter().filter()` can be combined into one.

Reply:
I've always wondered why Django lets you chain filters like this. Is there a performance difference between a single filter called and chained filter calls? I guess Django provides this interface to make long queries more legible but I don't know for sure.

Reply:
Looking at the [docs](https://docs.djangoproject.com/en/dev/topics/db/queries/#chaining-filters) it seems like chaining filters only makes sense if you need to filter on the intermediate QuerySet. Otherwise it's just as readable to put the kwargs on different lines.

Reply:
I would like to see the difference in the SQL that's generated, for science.

Reply:
I fixed these in this file. Thanks for the feedback.